python/ray/data/dataset.py

```diff
@@ -4557,6 +4558,9 @@ def materialize(self) -> "MaterializedDataset":
         )
         output._plan.execute()  # No-op that marks the plan as fully executed.
         output._plan._in_stats.dataset_uuid = self._get_uuid()
```

Comment:
do we need to update this as well?
`output._plan._in_stats.dataset_uuid = copy._get_uuid()`
also, should we instead use `output._set_uuid(<self/copy>._get_uuid())` for this and the following line to set the uuid of the underlying plan?

Reply:
yeah, I think we should be updating that as well. Also I think we can move the update of `output._plan._in_stats.dataset_uuid` into `Dataset._set_uuid` too.